node/rpc-api/src/data_types.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use async_std::channel::Sender;
+use async_std::sync::{Arc, RwLock};
+use beacon::BeaconEntry;
+use fil_types::SectorSize;
+use jsonrpc_v2::{MapRouter as JsonRpcMapRouter, Server as JsonRpcServer};
+use serde::{Deserialize, Serialize};
+
+use actor::market::{DealProposal, DealState};
+use address::{json::AddressJson, Address};
+use beacon::{json::BeaconEntryJson, Beacon, BeaconSchedule};
+use bitfield::json::BitFieldJson;
+use blocks::{
+    election_proof::json::ElectionProofJson, ticket::json::TicketJson,
+    tipset_keys_json::TipsetKeysJson, Tipset,
+};
+use blockstore::BlockStore;
+use chain::{headchange_json::SubscriptionHeadChange, ChainStore};
+use chain_sync::{BadBlockCache, SyncState};
+use cid::{json::CidJson, Cid};
+use clock::ChainEpoch;
+use fil_types::{json::SectorInfoJson, sector::post::json::PoStProofJson};
+use forest_libp2p::{Multiaddr, NetworkMessage};
+use ipld::json::IpldJson;
+use message::{
+    message_receipt::json::MessageReceiptJson, signed_message,
+    signed_message::json::SignedMessageJson, unsigned_message, SignedMessage, UnsignedMessage,
+};
+use message_pool::{MessagePool, MpoolRpcProvider};
+use num_bigint::{bigint_ser, BigInt};
+use state_manager::{MiningBaseInfo, StateManager};
+use vm::{ActorState, TokenAmount};
+use wallet::KeyStore;
+
+// RPC State
+#[derive(Serialize)]
+pub struct StreamingData<'a> {
+    pub json_rpc: &'a str,
+    pub method: &'a str,
+    pub params: SubscriptionHeadChange,
+}
+
+/// This is where you store persistent data, or at least access to stateful data.
+pub struct RPCState<DB, B>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    B: Beacon + Send + Sync + 'static,
+{
+    pub keystore: Arc<RwLock<KeyStore>>,
+    pub chain_store: Arc<ChainStore<DB>>,
+    pub state_manager: Arc<StateManager<DB>>,
+    pub mpool: Arc<MessagePool<MpoolRpcProvider<DB>>>,
+    pub bad_blocks: Arc<BadBlockCache>,
+    pub sync_state: Arc<RwLock<SyncState>>,
+    pub network_send: Sender<NetworkMessage>,
+    pub new_mined_block_tx: Sender<Arc<Tipset>>,
+    pub network_name: String,
+    pub beacon: Arc<BeaconSchedule<B>>,
+}
+
+#[derive(Serialize)]
+pub struct RPCSyncState {
```

Comment:
```suggestion
pub struct RpcSyncState {
```

Reply:
I actually changed this in this PR to the way I prefer it. We've disabled that linter rule anyway.